### PR TITLE
 feat: data BE + form validation

### PR DESCRIPTION
**Pull request by OstroumovAlex**
https://trello.com/c/g1zY0BhX/169-fe-add-be-data-to-internshipinfo
image

### CODE REVIEW

File: src/components/addProgram/AddProgram.jsx [hunk 1]
```diff
@@ -52,6 +54,11 @@ const checkDataReceived = (...arrays) =>
 
 const AddProgram = (props) => {
   const { closeModal } = props;
+  const { initialData } = props;
+  // const { dispatchFunction } = props;
```
Comment (nsukhotskaya) [inline src/components/addProgram/AddProgram.jsx:58]: Please delete unnecessary comment

File: src/components/addProgram/AddProgram.jsx [hunk 2]
```diff
@@ -110,21 +119,15 @@ const AddProgram = (props) => {
...
-      dispatch(fetchInternships());
+      newInternship.users = newInternship.users.map((user) => {
+        const userObject = {
+          ...allUsersList.filter((item) => item.userName === user),
+        };
+        return userObject;
+      });
       closeModal();
+      dispatch(updateFunction);
+      // dispatch(dispatchFunction(newInternship));
```
Comment (nsukhotskaya) [inline src/components/addProgram/AddProgram.jsx:130]: same

File: src/components/addProgram/AddProgram.jsx [hunk 3]
```diff
@@ -216,6 +219,14 @@ const AddProgram = (props) => {
...
+                    formik.touched[`${field.keyName}`] &&
+                    Boolean(formik.errors[`${field.keyName}`])
+                  }
+                  helperText={
+                    formik.touched[`${field.keyName}`] &&
+                    formik.errors[`${field.keyName}`]
+                  }
                 />
               ))}
               {/* eslint-disable react/jsx-props-no-spreading */}
```
Comment (nsukhotskaya) [inline src/components/addProgram/AddProgram.jsx:232]: Please fix eslint error

File: src/components/addProgram/AddProgram.jsx [hunk 4]
```diff
@@ -235,33 +246,46 @@ const AddProgram = (props) => {
                 </React.Fragment>
               ))}
               {/* eslint-enable react/jsx-props-no-spreading */}
```
Comment (nsukhotskaya) [inline src/components/addProgram/AddProgram.jsx:248]: same

File: src/utils/messages.json [hunk 5]
```diff
@@ -128,6 +129,8 @@
   "addprogram.field.label.allUsers": "Employees",
   "addprogram.input.date.format": "yyyy/MM/dd hh:mm a",
   "addprogram.input.date.mask": "___/__/__ __:__ _M",
+  "addprogram.validation.message.string":"This string should not be empty",
+  "addprogram.validation.message.array":"You should choose some item",
```
Comment (nsukhotskaya) [inline src/utils/messages.json:133]: I find this approach more friendly.
I recommend applying this in your application

image

**Comment by nsukhotskaya** [comment]
Use a red `*` to indicate to the user that the field is required

**Review by nsukhotskaya — APPROVED** [review]

**Review by ValeryiaMIRON — APPROVED** [review]

**Pull request merged**